check_diff/src/lib.rs

```diff
@@ -27,25 +28,46 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
 // One for the current master, and another for the feature branch
 // Parameters:
 // dest: Directory where rustfmt will be cloned
-pub fn compile_rustfmt(dest: &Path, inputs: CliInputs) -> Result<(), GitError> {
-    let clone_repo_result = clone_git_repo(RUSTFMT_REPO, dest);
+pub fn compile_rustfmt(dest: &Path, inputs: CliInputs) -> io::Result<Command> {
```

Comment:
I'm not sure that an `io::Result` makes sense here give that you're mapping errors to `Error::other`. You probably want to represent failures differently.

Also, my initial thought is that we shouldn't pass `CliInputs` to  `compile_rustfmt`. `compile_rustfmt` doesn't need to know that the arguments are being passed from the CLI. It would be better in my opinion if we passed the arguments like `rustfmt_repo` directly.

Reply:
i agree about the io::Result part. Have to think about it alittle more